### ISSUE TITLE
Student Enrolment - SuiteCRM to Moodle

### ISSUE THREAD

**Issue by thewebsitefactorylondon** [Myddleware/myddleware#72]
Hi Stéphane,

I hope you are well and hope you had a great weekend! I'd be grateful if you kindly suggest how to enrol students on Moodle through SuiteCRM. I have following scenario;
Students(Contacts) module in SuiteCRM to Users in Moodle [they are connected through Myddleware rule], this rule working perfectly fine. I have Course Enrolment module in SuiteCRM with following fields;

- Students (linked to Students/Contacts module)
- Enrolment Start Date
- Enrolment End Date
- Role ID (Student with value '5' - default value)
- Course ID (for this, I created a Dropdown with course id in Moodle)
_- User ID (I need help here how to achieve this)_

Now, I'd like to create another rule to enrol students on to different courses. I tried creating a rule for this, but it asked me to provide User ID in Moodle which was generated at the time of creating users (SuiteCRM to Moodle rule). How to get this or what is the easiest way to enrol students in Moodle through SuiteCRM?

Many thanks for your time and I look forward to hearing from you.
Best wishes,
Mohamad

**Comment by Myddleware** [comment]
Hi Mohamad,

I 'm well thanks. Hope you are well too. About what you want to acheive, you have to create a relationship between your first rule (user) and this new rue (enrolment). So when you create/edit your new rule, go to the tab "relations" : 
![relations](https://user-images.githubusercontent.com/11330301/35545492-a9fa450a-0545-11e8-8453-0c86c2921e65.png)

You should have something like "contact_id" in the list at left. Select it on the same row than "User ID". Then select your first rule (User) on the "Rules" column. 

With this relationship, Myddleware will read the contact ID in your enrolment module, search and the user ID corresponding to this contact ID in your first rule (User). Then Myddleware will create the enrolment with the User ID found.

Please let me know if it works for you.
Best regards,
Stéphane

**Comment by thewebsitefactorylondon** [comment]
Hi Stéphane,
Thanks for your detailed reply. Let me try this and get back to you.
Best wishes,
Mohamad

**Comment by thewebsitefactorylondon** [comment]
Hi Stéphane,
I tried above steps but I get an error after the transfer. But the good news is I was able to fetch the user ID linking with first rule(as per your suggestion). So one thing is sorted! : )

This is the error I get;

![image](https://user-images.githubusercontent.com/28488395/35569213-81f1d860-05c3-11e8-889a-3f5dd8e263d1.png)

![image](https://user-images.githubusercontent.com/28488395/35569231-903375dc-05c3-11e8-9fe5-2942ad9dd93c.png)

And here is what I have in Fields option

![image](https://user-images.githubusercontent.com/28488395/35569309-cc748db0-05c3-11e8-8fa3-a3f51f3d1871.png)

then in test data under simulation;

![image](https://user-images.githubusercontent.com/28488395/35569456-fc369408-05c3-11e8-80ff-83dad7650249.png)

and finally in the relationship option;

![image](https://user-images.githubusercontent.com/28488395/35569489-139e8e98-05c4-11e8-8483-3463d137aeb6.png)

I'd be grateful if you kindly let me know the solution.
Once again many thanks for your time and I look forward to hearing from you.

Best wishes,
Mohamad

**Comment by Myddleware** [comment]
Hi Mohamad,
I guess your problem is the date format, Moodle need a timestamp : 
![timestamp_moodle](https://user-images.githubusercontent.com/11330301/35588788-0f5c12b0-05d8-11e8-98d9-2ccab01bd733.png)

So, you can create a formula to transform you dates in timestamp, somethig like that : 
changeFormatDate( {course_end_date},"Y-m-d","U")

You can see this article to know more about formula : https://myddleware.github.io/myddleware/#/advanced_usage?id=formulae

Please let me know if it works.
Stéphane

**Comment by thewebsitefactorylondon** [comment]
Hi Stéphane,

I hope you are well and thank you for providing this solution for the course date. My apologies for getting back to you late. I was away this week and did not get a chance to try this fix. 

This fixed the error. 👍 I created two rules (one for creating courses and one for course enrolment where I linked course and user rule relationship option) and followed your instructions. It worked like a charm! Once again thank you so much for your continued support. Have a great weekend! : )

Best wishes,
Mohamad
